service/src/main/java/com/epam/esm/service/service/user/UserServiceImpl.java

```diff
@@ -42,13 +46,22 @@ public UserServiceImpl(UserDAO dao,
 
     @Override
     @Transactional
-    public User create(User user) {
-        throw new UnsupportedOperationException();
+    public User create(User user) throws ValidationException, BadRequestException {
+        validator.validate(user);
+        try {
+            if (getByUniqueLoginOptional(user.getLogin()).isPresent()) {
+                throw new BadRequestException("Such login already exists! login= " + user.getLogin(),
+                        ErrorCodes.USER_BAD_REQUEST);
+            }
+            return dao.create(user);
+        } catch (DataIntegrityViolationException e) {
+            throw new BadRequestException(e, ErrorCodes.USER_BAD_REQUEST);
+        }
     }
 
     @Override
-    public User read(int id) throws NotFoundException {
-        Optional<User> userOptional = readOptional(id);
+    public User getById(int id) throws NotFoundException {
```

Comment:
change type to Optional<User> and remove the second method

Reply:
need to discuss

Reply:
Done